component/src/main/java/org/wso2/extension/siddhi/io/feed/sink/FeedSink.java

```diff
@@ -0,0 +1,268 @@
+                Constants.FEED_CREATE));
+
+        int timeout = validateTimeOut();
+        this.abderaClient.setConnectionTimeout(timeout);
+        String userName = optionHolder.validateAndGetStaticValue(Constants.USERNAME, Constants.CREDENTIALS);
+        String password = optionHolder.validateAndGetStaticValue(Constants.PASSWORD, Constants.CREDENTIALS);
+        if (!optionHolder.validateAndGetStaticValue(Constants.USERNAME, Constants.CREDENTIALS)
+                .equals(Constants.CREDENTIALS)
+                || !optionHolder.validateAndGetStaticValue(Constants.PASSWORD, Constants.CREDENTIALS)
+                        .equals(Constants.CREDENTIALS)) {
+            AbderaClient.registerTrustManager();
+            try {
+                abderaClient.addCredentials(String.valueOf(url), Constants.REALM, Constants.AUTH_SCHEME,
+                        new UsernamePasswordCredentials(userName, password));
+            } catch (URISyntaxException e) {
+                // nothing, url syntax exception doesn't throw because url string getting form a URL object
+            }
+        }
+    }
+
+    // atom function validation
+    private String validateAtom(String atomFunc) {
+        atomFunc = atomFunc.toLowerCase(Locale.ENGLISH);
+        if (atomFunc.equals(Constants.FEED_UPDATE)
+                || atomFunc.equals(Constants.FEED_DELETE) || atomFunc.equals(Constants.FEED_CREATE)) {
+            return atomFunc;
+        }
+        throw new SiddhiAppValidationException("Atom function validation error  in " + streamDefinition.getId()
+                + ". Acceptance parameters are 'create', 'delete', 'update'");
+    }
+
+    private int validateTimeOut() {
+        try {
+            int timeout = Integer.parseInt(optionHolder.validateAndGetStaticValue(Constants.TIME_OUT,
+                    Constants.DEFAULT_TIME_OUT));
+            if (timeout > 0) {
+                return timeout;
+            } else {
+                throw new SiddhiAppValidationException("Error in " + streamDefinition.getId()
+                        + " validating timeout, Response timeout accept only positive integers. But found " +
+                        optionHolder.validateAndGetStaticValue(Constants.TIME_OUT,
+                                Constants.DEFAULT_TIME_OUT));
+            }
+        } catch (NumberFormatException e) {
+            throw new SiddhiAppValidationException("Error in " + streamDefinition.getId()
+                    + " validating timeout, Response timeout accept only positive integers. But found " +
+                    optionHolder.validateAndGetStaticValue(Constants.TIME_OUT,
+                            Constants.DEFAULT_TIME_OUT));
+        }
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions dynamicOptions) throws ConnectionUnavailableException {
+        HashMap<String, String> map = (HashMap) payload;
+        ClientResponse resp = null;
+        switch (atomFunc) {
```

Comment:
Have the cases in Constants in Uppercase and within switch set atomFunc.toUpperCase()